src/DRICH_geo.cpp

```diff
@@ -546,29 +574,94 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
+	    // if(isec==0) printf("%d %f %f\n",imod,sensorPV.position().x(),sensorPV.position().y());
 
-          // properties
-          sensorPV.addPhysVolID("sector", isec)
+	    // properties
+	    sensorPV.addPhysVolID("sector", isec)
               .addPhysVolID("module", imod); // NOTE: must be consistent with `sensorIDfields`
-          auto        imodsec    = encodeSensorID(sensorPV.volIDs());
-          std::string modsecName = secName + "_" + std::to_string(imod);
-          DetElement  sensorDE(det, "sensor_de_" + modsecName, imodsec);
-          sensorDE.setPlacement(sensorPV);
-          if (!debugOptics || debugOpticsMode == 3) {
-            SkinSurface sensorSkin(desc, sensorDE, "sensor_optical_surface_" + modsecName, sensorSurf, sensorVol);
-            sensorSkin.isValid();
-          };
-
-          // increment sensor module number
-          imod++;
-
-        }; // end patch cuts
-      };   // end phiGen loop
-    };     // end thetaGen loop
+	    auto        imodsec    = encodeSensorID(sensorPV.volIDs());
+	    std::string modsecName = secName + "_" + std::to_string(imod);
+	    DetElement  sensorDE(det, "sensor_de_" + modsecName, imodsec);
+	    sensorDE.setPlacement(sensorPV);
+	    if (!debugOptics || debugOpticsMode == 3 || debugOpticsMode == 5) {
+	      SkinSurface sensorSkin(desc, sensorDE, "sensor_optical_surface_" + modsecName, sensorSurf, sensorVol);
+	      sensorSkin.isValid();
+	    };
+
+	    // increment sensor module number
+	    imod++;
+
+	  }; // end patch cuts
+	};   // end phiGen loop
+      };     // end thetaGen loop
+    }; // end sensor loop
+
+    // large sensor placement (for focal point testing):
+    if (debugOpticsMode == 4){
+      Box    sensorSolidScaled(sensorRescale * sensorSide / 2., sensorRescale * sensorSide / 2., sensorThickness / 2.);
+      Volume sensorVolScaled(detName + "_sensor_" + secName, sensorSolidScaled, sensorMat);
+      sensorVolScaled.setVisAttributes(sensorVis);
+      sensorVolScaled.setSensitiveDetector(sens);
+
+      double phiGen = 0;
+      auto sensorPlacement =
+        RotationZ(sectorRotation) *
+        Translation3D(sensorSphPos.x()+100, sensorSphPos.y(), sensorSphPos.z()-125) *
+        RotationX(phiGen) *
+        Translation3D(sensorSphRadius, 0., 0.)*
+        RotationZ(0);
+      auto sensorPV = gasvolVol.placeVolume(sensorVolScaled, sensorPlacement);
+      sensorPV.addPhysVolID("sector", isec).addPhysVolID("module", imod);
+
+      DetElement sensorDE(det, Form("sensor_de%d_%d", isec, imod), 0);
+      sensorDE.setPlacement(sensorPV);
+
+      SkinSurface sensorSkin(desc, sensorDE, Form("sensor_optical_surface%d", isec), sensorSurf, sensorVolScaled);
+      sensorSkin.isValid();
+    }; // end large sensor placement
+
+    // for debugOpticsMode 5: drawing calculated focal points
+    // of thrown beams of parallel photons
+    if( std::getenv("FPPLOT_FILE")!=NULL && debugOpticsMode == 5){
```

Comment:
Please avoid environment variables as an interface to specifying information. It is not manifest when running and leads to problems with reproducibility of results. Suggestion is to create a setting in the xml interface so the file name is stored.